epictrack-api/src/api/services/work.py

```diff
@@ -674,3 +583,96 @@ def check_work_nation_existence(
         if query.count() > 0:
             return True
         return False
+
+    @classmethod
+    def handle_phase(cls, work_phase, phase_event_templates):  # pylint: disable=too-many-locals
```

Comment:
nice work for getting a lot of things done..good job..
one improvement i can suggest is to split this method in multiple smaller methods to improve maintainability ? this is specifically important since we dont have lot of tests ..so whoever is going to maintain this might face a hard time to make changes..
Not for this particular PR ;but great if we can 
1) Write comments ..a lot of explain logic
2) try to split this into multiple methods